TRIAGE.md

```diff
@@ -0,0 +1,49 @@
+# Triage in OpenSearch Core
+## What Do We Do With These Labels Anyway?
+
+Across all repositories in opensearch-project, we have a process called “triage.” The goal of this process is to make sure repository owners and maintainers check new issues to understand them. First - we learn if the issue is security related (e.g. CVE), second - we understand if we have enough information to act on the issue, third - we re-label or respond on the issue. During triage we do not work on these issues, but instead just try to get the issue to a next step. Because the OpenSearch core repository (https://github.com/opensearch-project/OpenSearch) has so many different functions with different sets of contributors working on these functions, repo owners decided to treat these functions like separate repositories when triaging to get through the backlog faster and have more focus. There are a few maintainers that do triage in public (see https://meetup.com/OpenSearch to join) and there will be more coming.
+
+Here is an example of the workflow for a new issue in the core repo.
```

Comment:
I think there should be a body of maintainers/contributors that agreeing to work in this way, not a generic example.

Note; I'm not part of any such team for core, so I triage on my own whims and timescales, otherwise I'd volunteer.

Reply:
Open to whatever the maintainers/other contributors want to do here. I haven't seen any other objections to date so this example remains. Can you suggest something more concrete? Not sure I understand what you mean. 

Thanks for taking on this review!

Reply:
Any luck finding a first follower(s) on this topic? I am concerned about documenting a process in a document that no one follows.

Reply:
I agree. I'll post this to the Slack #maintainers channel and ask for folks to review this PR.

Reply:
Personally, I side with @peternied here - we should leave the process(es) and workflow(s) up to repository maintainers to decide. The pace of changes varies greatly between repositories.

Reply:
I absolutely agree. No point in pushing a proposal that will be ignored. What's more important is the discussion and resolution. For clarity, @reta, this proposal is specific to the OpenSearch repo. 

Here's the challenge we have right now: There are 80 untriaged issues in the OpenSearch repo. There are >73 that were created more than 2 weeks ago: https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch+label%3Auntriaged+is%3Aopen+created%3A%3C2023-12-19+is%3Aissue&type=issues. 

The triaging that's happening now in core isn't enough, IMO. Do maintainers agree?  

If so, how can the community help maintainers drive this down? Is there a better process to propose that maintainers can agree on?

Reply:
> The triaging that's happening now in core isn't enough, IMO. Do maintainers agree?

That is very fair, we don't have triage meetings for core specifically but we should I think

Reply:
Thanks @reta. Are we back to the idea of a core public triage meeting, then? It may make sense to develop the process from there and maintainers can document the process used and decide on useful labels with the community there. @peternied, @reta, I can help set something like that up with you (I'm not a core maintainer so I would be open to facilitating and recommending, but the maintainers should determine the process).

Reply:
@macohen You're trying to document the process that is currently being done, right? Much of this happens internally within Amazon teams, but some of it happens in open community meetings, but to date the actually process isn't documented anywhere in the open. Is that right?

Reply:
> Thanks @reta. Are we back to the idea of a core public triage meeting, then? I

I would love to see that happening (it is going well for security / ml-commons / search relevance) but I do not see (personally, as of today) how we could make it work: there are many components (or subsystems) in there, search relevance is an example of cutting a small piece of the core and looking into it, but there are many others pieces. 

Having one meeting to go over the OpenSearch core is an option (we need at least few leads to be present there) - it may be efficient use of maintaners' time but not users times. Having N meetings (one per component or subsystem) is another option that has own pros and cons. 

Depending on the way we would like to approach the problem, the triaging rules may differ.

Reply:
> @macohen You're trying to document the process that is currently being done, right? Much of this happens internally within Amazon teams, but some of it happens in open community meetings, but to date the actually process isn't documented anywhere in the open. Is that right?

Right. I don't know what the process is across all repos and I think the process has recently become more confusing and unclear so I'm attempting a declarative document that says what should be. If it's the wrong declaration, then let's change the document, but let's document the right thing. And let's keep this specific to OpenSearch repo since that's where the complexity is. I would also say that we need to make this process public and I am even more certain that the maintainers should own it.